tools/explorer/tt_adapter/src/tt_adapter/mlir.py

```
@@ -932,8 +938,8 @@ def build_graph(
 
     # Check if all perf locations match some graph node
     for loc in loc_to_perf.keys():
-        pass  # https://github.com/tenstorrent/tt-mlir/issues/3255
-        # assert loc in processed_locs, f"Perf location {loc} not found in graph nodes"
+        if loc in processed_locs:
+            logging.error(f"Perf location {loc} not found in graph nodes")
```

What @nsmithtt said makes sense, we don't want to be terminating the server. I think we could raise an exception here, the frontend handles this and reports the error to the user without the server crashing? @vprajapati-tt

---

Hey sorry, this got lost in my inbox. I agree, we should aim to increase telemetry instead of just crashing.

---

If an error is raised we do show that off as an error message, with the logs being preserved. I noticed a while back that if an error is observed than data can be lost if a packet was "in transit". A "gentle" failure method is something we can add into explorer, I added an issue to track this @ #3516